remote_test.go

```diff
@@ -198,7 +296,16 @@ func (s *RemoteSuite) testFetch(c *C, r *Remote, o *FetchOptions, expected []*pl
 	c.Assert(err, IsNil)
 	l.ForEach(func(r *plumbing.Reference) error { refs++; return nil })
 
-	c.Assert(refs, Equals, len(expected))
+	if o.Depth > 0 {
+		shallowCommits, err := r.s.Shallow()
+		c.Assert(err, IsNil)
+
+		// Only the depth-most parent of any given commit will be shallow.
+		commits := len(r.s.(*memory.Storage).Commits) / o.Depth
```

Comment:
This calculation isn't perfect. It fails to account for some more complicated scenarios.